Makefile

```diff
@@ -1,15 +1,55 @@
-.PHONY: clean all debug test
+.PHONY: all clean env test bench gox test-integration
 
-all:
-	for dir in ./cmd/* ; do \
-		(echo "$$dir"; cd "$$dir"; go build) \
-	done
+TMPGOPATH=$(PWD)/.gopath
+VENDORPATH=$(PWD)/Godeps/_workspace
+BASE=github.com/restic/restic
+BASEPATH=$(TMPGOPATH)/src/$(BASE)
 
-debug:
-	(cd cmd/restic; go build -a -tags debug)
+GOPATH=$(TMPGOPATH):$(VENDORPATH)
 
-test:
-	./testsuite.sh
+GOTESTFLAGS ?= -v
+GOX_OS ?= linux darwin openbsd freebsd
+SFTP_PATH ?= /usr/lib/ssh/sftp-server
+
+export GOPATH GOX_OS
+
+all: restic
+
+.gopath:
+	mkdir -p .gopath/src/github.com/restic
+	ln -sf ../../../.. .gopath/src/github.com/restic/restic
```

Comment:
This is the structure after `.gopath` has been created:
```
$ tree .gopath 
.gopath
└── src
    └── github.com
        └── restic
            └── restic -> ../../../..
```

And `GOPATH` is set to the following value, this can be seen with `make env`:
```
$ make env
export GOPATH="/home/fd0/tmp/restic/.gopath:/home/fd0/tmp/restic/Godeps/_workspace"
```

So when the go tool is run and encounters the path `github.com/restic/restic/chunker`, the following path is constructed from the first element of `GOPATH`:
```
/home/fd0/tmp/restic/.gopath/src/github.com/restic/restic/chunker
```

Since `restic` is a symlink, it is resolved to:
```
/home/fd0/tmp/restic/.gopath/src/github.com/restic/../../../../chunker
```

Which is the same as:
```
/home/fd0/tmp/restic/chunker
```

That's why it is `../../../..`. Tadaa!